### ISSUE TITLE
[Discussion] Markov Decision Processes

### ISSUE THREAD

**Issue by czgdp1807** [sympy/sympy#19257]
https://math.la.asu.edu/~jtaylor/teaching/Fall2012/STP425/lectures/MDP.pdf summarises Markov decision processes in chapter 3 and chapter 4 has some use cases. There are some symbolic aspects of decision processes which I think can be added to SymPy, for example computing expected reward at the end of decision epochs.

P.S. We don't have to discuss the coding part of MDPs initially. Comments are invited from the community for deciding the feasibility of this idea.

**Comment by Smit-create** [comment]
I read about MDP and seems interesting as it involves both deterministic and randomized(probability distributions). I think the concept of reward and choosing actions in MDP are probably the Reinforcement learning concepts(I have not read much about it). Also, if they can be added to sympy, then we can think of adding Markovian deterministic and randomized, but for history-dependent deterministic and randomized, it will be harder to implement as we need to check for all the actions and states of the pasts.

**Comment by czgdp1807** [comment]
> I think the concept of reward and choosing actions in MDP are probably the Reinforcement learning concepts(I have not read much about it)

There was a result given in the above document which helps in finding the maximum expected reward directly under certain conditions. May be we can use it. Let's start by picking an example from the above document and solve it using some fake APIs and some from sympy. Showing a fake python session would be okay.

**Comment by mridul7ahuja** [comment]
Yes, some of the results for optimal decision problems and even for policy evaluation (if not determination like in value iteration) can be directly used and might be very useful! I have previously worked a lot with MDPs in Python and would definitely be interested to help. Please let me know if any progress was made earlier and how I can contribute.
